cpu/cc2538/clock.c

```diff
@@ -62,7 +62,21 @@
 #include <stdint.h>
 /*---------------------------------------------------------------------------*/
 #define RTIMER_CLOCK_TICK_RATIO (RTIMER_SECOND / CLOCK_SECOND)
-#define RELOAD_VALUE (125000 - 1)  /** Fire 128 times / sec */
+
+/* Prescaler for GPT0:Timer A used for clock_delay_usec(). */
+#if SYS_CTRL_SYS_CLOCK < SYS_CTRL_1MHZ
+#error System clock speeds below 1MHz are not supported
+#else
+#define PRESCALER_VALUE     (SYS_CTRL_SYS_CLOCK / 1000000 - 1)
+#endif
+
+/* Reload value for SysTick counter */
+#if SYS_CTRL_SYS_CLOCK < SYS_CTRL_2MHZ
+/* Clock speeds below 2 MHz will lead to reduced accurracy */
+#error System clock speeds below 2MHz are not supported
```

Comment:
Actually, it's 2 MHz now only because `CLOCK_SECOND` is 128, but `CLOCK_CONF_SECOND` could be changed, which would make this test wrong. So the generic test would be:
```
#if SYS_CTRL_SYS_CLOCK % CLOCK_SECOND
#error System clock speed too slow for CLOCK_SECOND, leading to reduced accuracy
```